src/util/ivmCache/contextReset.js

```diff
@@ -0,0 +1,226 @@
+        reject.applyIgnored(undefined, [new ivm.ExternalCopy(err).copyInto()]);
+      } finally {
+        stats.timing('fetchV2_call_duration', fetchStartTime, fetchTags);
+      }
+    }),
+  );
+
+  await jail.set(
+    '_geolocation',
+    new ivm.Reference(async (resolve, reject, ...args) => {
+      const geoStartTime = new Date();
+      const geoTags = { ...trTags };
+      try {
+        if (args.length === 0) {
+          throw new Error('ip address is required');
+        }
+        if (!process.env.GEOLOCATION_URL) throw new Error('geolocation is not available right now');
+        const res = await fetch(`${process.env.GEOLOCATION_URL}/geoip/${args[0]}`, {
+          timeout: GEOLOCATION_TIMEOUT_IN_MS,
+        });
+        if (res.status !== 200) {
+          throw new Error(`request to fetch geolocation failed with status code: ${res.status}`);
+        }
+        const geoData = await res.json();
+        geoTags.isSuccess = 'true';
+        resolve.applyIgnored(undefined, [new ivm.ExternalCopy(geoData).copyInto()]);
+      } catch (error) {
+        const err = JSON.parse(JSON.stringify(error, Object.getOwnPropertyNames(error)));
+        geoTags.isSuccess = 'false';
+        reject.applyIgnored(undefined, [new ivm.ExternalCopy(err).copyInto()]);
+      } finally {
+        stats.timing('geo_call_duration', geoStartTime, geoTags);
+      }
+    }),
+  );
+
+  await jail.set('_getCredential', (key) => {
+    if (isNil(credentials) || !isObject(credentials)) {
+      logger.error(
+        `Error fetching credentials map for transformationID: ${cachedIsolate.transformationId} and workspaceId: ${cachedIsolate.workspaceId}`,
+      );
+      stats.increment('credential_error_total', trTags);
+      return undefined;
+    }
+    if (key === null || key === undefined) {
+      throw new TypeError('Key should be valid and defined');
+    }
+    return credentials[key];
+  });
+
+  await jail.set('extractStackTrace', (trace, stringLiterals) =>
+    extractStackTraceUptoLastSubstringMatch(trace, stringLiterals),
+  );
+}
+
+/**
+ * Reset the context of a cached isolate for fresh execution
+ * @param {Object} cachedIsolate The cached isolate object
+ * @param {Object} credentials Fresh credentials for this execution
+ * @returns {Object} Cached isolate with reset context ready for execution
+ */
+async function resetContext(cachedIsolate, credentials = {}) {
+  if (!cachedIsolate?.isolate) {
+    throw new Error('Invalid cached isolate provided for context reset');
+  }
+
+  try {
+    // Create a new context for this execution
+    const newContext = await cachedIsolate.isolate.createContext();
+    const jail = newContext.global;
+
+    // Set up global object properly
+    await jail.set('global', jail.derefInto());
+
+    // Re-inject the required APIs with fresh state
+    await injectFreshApis(jail, cachedIsolate, credentials);
+
+    // Set up bootstrap script in the new context
+    await cachedIsolate.bootstrap.run(newContext);
```

Comment:
don't understand where bootstrap is set like it's set [here](https://github.com/rudderlabs/rudder-transformer/blob/b50b7494a6d774813ee95a9ed3b1998cd70b3326/src/util/customTransformer.js#L168)
may be I missed something.
Also I see a opportunity to bootstrap only once and use the same bootstrap for all contexts. ie compile once and use cache it per isolate and use it for every context. 
That means, we can refer to same main isolate function references. instead of creating one per context. let me know your thoughts.